### PR TITLE
[KYUUBI #3764] Introduce the base parser framework for kyuubi server

### PR DESCRIPTION
**Pull request by Yikf**
### _Why are the changes needed?_

Close https://github.com/apache/incubator-kyuubi/issues/3764, This pr aims to add the base parser framework to kyuubi server for the following purposes:
- Add syntax that is executable on the server side to enhance capabilities
- Add the ability to translate SQL

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by wForget** [comment]
I prefer syntax like iceberg spark procedures:

```
CALL  func(params)
```

FYI: #1301

File: kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4 [hunk 1]
```diff
@@ -0,0 +1,178 @@
+/*
```
Comment (yaooqinn) [inline kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4:1]: can we separate Lexer and Parser?

File: kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4 [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    ;
+
+runnable
+    : KILL ENGINE                                                   #killEngine
+    | LAUNCH ENGINE                                                 #launchEngine
+    | ALTER SESSION SET propertyList                                #alterEngineConfig
+    ;
+
+translate
+    : RENAME TABLE identifier TO identifier                         #renameTable
```
Comment (pan3793) [inline kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4:65]: we should be careful about extending the syntax, currently, the generic Kyuubi SQL scope is narrowed to management. it says, the typical ANSI-like SQL is 

```
SHOW <RESOURCES>
CREATE <RESOURCE> ...
ALTER <RESOURCE> SET ...
DROP <RESOURCE> ...
```

In Kyuubi case, I think we can start w/ the resource `SESSION` or `ENGINE`.


The "translate" should be protocol-specific, e.g. we only need it for MySQL FE now.

File: kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4 (hunk 1 above)
Comment (Yikf) [reply kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4:1]: Thanks, will separate Lexer and Parser files

File: kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4 (hunk 2 above)
Comment (Yikf) [reply kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4:65]: Thanks for your thoughts, That's great; and thanks @wForget, prefer this style to that of `CALL  func(params)`, what do you think~

@pan3793 I like this style for extending the syntaxm, but and have a bit weird if we use `CREATE ENGINE` for launch engine, `DROP ENGINE` for kill/stop engine.

File: kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4 [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+statement
+    : runnable                                                      #logicalRunnable
+    | translate                                                     #logicalTranslate
+    | .*?                                                           #passThrough
+    ;
+
+runnable
+    : KILL ENGINE                                                   #killEngine
+    | LAUNCH ENGINE                                                 #launchEngine
+    | ALTER SESSION SET propertyList                                #alterEngineConfig
```
Comment (cfmcgrady) [inline kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4:61]: then the `ALTER` DDL still works? for instance:

```sql
ALTER TABLE loc_orc SET fileformat orc;
```

**Pull request closed without merging**